composer/trainer/trainer.py

```diff
@@ -232,6 +234,16 @@ def _is_cuda_oom(e: RuntimeError):
     return False
 
 
+def _fsdp_reshard_and_cleanup(model: torch.nn.Module):
+    for name, module in model.named_modules():
+        if isinstance(module, FullyShardedDataParallel):
+            if module.check_is_root():
+                try:
+                    _post_backward_final_callback(module, module)
+                except Exception as e:
+                    log.warning(f'Failed to reshard fsdp after oom, error: {e}')
```

Comment:
Have you ever seen this?

Reply:
No, i think it should be safe to remove the try catch.

Reply:
Yeah I think if this fails it should probably just fail the run, not just log?